clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -6473,6 +7081,31 @@ run_lockstep_simulation_for_kernel_seq(scheduler_t &scheduler, int num_outputs,
                 continue;
             }
             assert(status == scheduler_t::STATUS_OK);
+
+            // Tests for get_next_trace_pc().
+            uint64_t pc;
+            if (
+                // We expect the returned next trace pc to change when the decision to
+                // inject a syscall or switch trace is made.
+                // XXX: Could we avoid this and do some read-ahead so we can provide the
+                // actual next pc when returning the prior user-space instr? Perhaps
+                // doable for syscalls but not so easy for context switches as that
+                // would mean preponing the context switch decision before it needs to
```

Comment:
"preponing"? I have never seen this before and didn't think it was a word: searching says it's used in Indian English.  Hmm.